### PR TITLE
client: Add RmlElement class

### PR DESCRIPTION
**Pull request by xDavidos**
You'd better finish this, I understand that not everything is ready for RmlElements, or I don't understand something.

### CODE REVIEW

**Comment by LeonMrBonnie** [comment]
You should remove all the functions related to events, those are done in JS. (I already wrote a system for that, you can check how alt.Audio uses it)

And I don't see whats missing for all the stuff you didn't add? You can use e.g. this so a user can pass a RML element to the method:
```cpp
alt::IRmlElement* element;
if(!ctx.GetArg(0, element)) return;
```
That will automatically do everything necessary such as checking the type etc.

File: client/src/classes/RmlElement.cpp [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    // TODO
+}
+
+// clang-format off
+extern js::Class baseObjectClass;
+extern js::Class rmlElementClass("RmlElement", &baseObjectClass, nullptr, [](js::ClassTemplate& tpl)
+{
+    tpl.BindToType(alt::IBaseObject::Type::RML_ELEMENT);
+
+    tpl.Property<&alt::IRmlElement::GetRelativeOffset>("relativeOffset");
```
[the remark this answers is not in the document]
Comment (LeonMrBonnie) [reply client/src/classes/RmlElement.cpp:83]: No, leave those like that, those are properties so they shouldn't have `get` in the beginning

**Comment by LeonMrBonnie** [comment]
For what do you need `GetBaseObjectOrNull`? Thats always done automatically in v2 and I don't know where you would need it.

For the custom object take a look at how e.g. `syncedMeta` property for Entity class works.

**Comment by xDavidos** [comment]
I hope I understood everything correctly and used manually bindings where it is really needed😉

File: client/src/classes/RmlElement.cpp [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+    tpl.Property<&alt::IRmlElement::GetAbsoluteOffset>("absoluteOffset");
+    tpl.Property<&alt::IRmlElement::GetBaseline>("baseline");
+    tpl.Property<&alt::IRmlElement::GetZIndex>("zIndex");
+    tpl.Property<&alt::IRmlElement::GetContainingBlock>("containingBlock");
+    tpl.Property<&alt::IRmlElement::GetFocusedElement>("focusedElement");
+    tpl.Property<&alt::IRmlElement::GetTagName>("tagName");
+    tpl.Property<&alt::IRmlElement::GetRmlID, &alt::IRmlElement::SetRmlID>("rmlId");
```
Comment (LeonMrBonnie) [inline client/src/classes/RmlElement.cpp:171]: `rmlId` should be `rmlID`

**Pull request merged**